test/unittests/commands/ingest_xmlupload/test_apply_ingest_uuid.py

```diff
@@ -0,0 +1,72 @@
+from lxml import etree
+
+from dsp_tools.commands.ingest_xmlupload.apply_ingest_uuid import replace_filepath_with_sipi_uuid
+
+# pylint: disable=missing-class-docstring,missing-function-docstring
+
+
+class TestReplaceBitstreamPaths:
+    def test_all_good(self) -> None:
+        xml = (
+            "<knora>"
+            '<resource label="Fluffy Image" restype=":Image2D" id="Fluffy1" permissions="res-default">'
+            '<bitstream permissions="prop-default">images/Fluffy.jpg</bitstream>'
+            "</resource>"
+            "</knora>"
+        )
+        root = etree.ElementTree(etree.fromstring(xml))
+        reference_dict = {"images/Fluffy.jpg": "fluffy_uuid"}
+        res_tree, res_msg = replace_filepath_with_sipi_uuid(root, reference_dict)
+        assert not res_msg.media_no_uuid
+        assert not res_msg.unused_media_paths
+        expected = (
+            b"<knora>"
+            b'<resource label="Fluffy Image" restype=":Image2D" id="Fluffy1" permissions="res-default">'
+            b'<bitstream permissions="prop-default">fluffy_uuid</bitstream>'
+            b"</resource></knora>"
+        )
+        assert etree.tostring(res_tree) == expected
```

Comment:
I feel this kind of thing is better tested directly on the xml tree, rather than comparing serialization:  
Your assertion doesn't really care if the strings are the same, you really only care thet the XPath `./knora/resource/bitstream/text()` no longer contains "images/Fluffy.jpg", but rather "fluffy_uuid".  
I don't recall the syntax from the top of my head, but there are at least 3 ways (object navigation, xpath, css selectors) to get to a certain place in an lxml tree; and the node should give you its text; and then you can just compare the string values.  
(you could even generate the xml tree in lxml rather than from a string, but I'm not sure how nice that would look/feel)